internal/castai/castai.go

```diff
@@ -96,9 +98,25 @@ func createHTTPTransport() *http.Transport {
 		IdleConnTimeout:       90 * time.Second,
 		TLSHandshakeTimeout:   5 * time.Second,
 		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig:       setTLSConfig(log),
 	}
 }
 
+func setTLSConfig(log logrus.FieldLogger) *tls.Config {
```

Comment:
I suggest avoid using `log` param here: instead return an error and log it after calling